### PR TITLE
[CQT-230] Create SWAP to CNOT decomposer

### PR DESCRIPTION
**Pull request by elenbaasc**
- Implement SWAP2CNOT decomposer
- Add tests for SWAP2CNOTdecomposer
- Add SWAP and SWAP2CNOT decomposer to spin qubit integration test and update output

### CODE REVIEW

**Review by rturrado — COMMENTED** [review]
Beautiful!

Same question as for the CNOT2CZ decomposer. Do we want to update any docs? Like, specifying the list of available decomposers?

**Comment by elenbaasc** [comment]
> Beautiful!
> 
> Same question as for the CNOT2CZ decomposer. Do we want to update any docs? Like, specifying the list of available decomposers?

Yes! I put it as a TODO in the description, so I still have some work to do 😋

**Comment by rturrado** [comment]
> Yes! I put it as a TODO in the description, so I still have some work to do 😋

Ah cool. Sorry, didn't read that.

**Review by rturrado — COMMENTED** [review]
Good, thanks!

I see you have changed things like CNOT to `CNOT`. I would take the opportunity to do that accross the whole `tutorial.md`. But this can be done as part of another PR.

I would strongly advise for revising the whole structure of `tutorial.md`: add new `md` files, one per pass, or something like that.

File: docs/tutorial.md [hunk 1]
```diff
@@ -369,6 +369,13 @@ _Output_:
 
     replacement for gate CNOT does not preserve the quantum state
 
+##### _SWAP to CNOT decomposer_
```
[the remark this answers is not in the document]
Comment (elenbaasc) [reply docs/tutorial.md:372]: Good point! I've made a new ticket to revise the tutorial. I will add these notes in the description.

**Review by rturrado — APPROVED** [review]

**Pull request merged**